src/ngraph/op/reverse.cpp

```diff
@@ -69,3 +70,116 @@ void op::Reverse::generate_adjoints(autodiff::Adjoints& adjoints, const NodeVect
 
     adjoints.add_delta(x, make_shared<op::Reverse>(delta, m_reversed_axes));
 }
+
+const string op::v1::Reverse::type_name{"Reverse"};
+
+op::v1::Reverse::Reverse(const Output<Node>& data,
+                         const Output<Node>& reversed_axes,
+                         const std::string& mode)
+    : Op({data, reversed_axes})
+    , m_mode{mode}
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::v1::Reverse::validate_and_infer_types()
+{
+    NODE_VALIDATION_CHECK(this,
+                          m_mode == "index" || m_mode == "mask",
+                          "The provided value of the 'mode' attribute (",
+                          m_mode,
+                          ") is invalid. Allowed values: 'index' or 'mask'.");
+
+    if (m_mode == "mask")
+    {
+        NODE_VALIDATION_CHECK(this,
+                              get_input_element_type(1) == element::boolean,
+                              "In 'mask' mode the second input must contain boolean values.");
+    }
+
+    const auto input_shape = get_input_partial_shape(0);
+    const auto input_rank = input_shape.rank();
+
+    const auto rev_axes_shape = get_input_partial_shape(1);
+    const auto rev_axes_rank = rev_axes_shape.rank();
+
+    if (input_rank.is_static() && rev_axes_rank.is_static())
+    {
+        NODE_VALIDATION_CHECK(this,
+                              static_cast<size_t>(rev_axes_rank) == 1,
+                              "The reversed_axes input must be a 1D tensor (got ",
+                              static_cast<size_t>(rev_axes_rank),
+                              ").");
+
+        if (m_mode == "mask")
+        {
+            NODE_VALIDATION_CHECK(
+                this,
+                static_cast<size_t>(rev_axes_rank) == 1,
+                "The reversed_axes input rank needs to be equal to 1 in 'mask' mode.");
+
+            const auto rev_axes_mask_elems_count = static_cast<size_t>(rev_axes_shape[0]);
+            NODE_VALIDATION_CHECK(this,
```

Comment:
`input_rank` is checked only in this `VALIDATION_CHECK` so maybe you should add here a new `if` and move `input_rank.is_static()` condition from line 106 to this `if`?

Reply:
Thanks, I've moved the the `input_rank` check deeper and realized that I also have to check if the first(or zero) dimension of the shape is static before I cast it to `size_t`.